src/main/java/org/eclipse/uprotocol/client/usubscription/v3/InMemoryUSubscriptionClient.java

```diff
@@ -39,45 +41,84 @@
+    private static final UUri REGISTER_NOTIFICATIONS_METHOD = UriFactory.fromProto(USUBSCRIPTION, 6);
+    private static final UUri UNREGISTER_NOTIFICATIONS_METHOD = UriFactory.fromProto(USUBSCRIPTION, 7);
+
+    private static final UUri NOTIFICATION_TOPIC = UriFactory.fromProto(USUBSCRIPTION, 0x8000);
+
 
     // Map to store subscription change notification handlers
     private final ConcurrentHashMap<UUri, SubscriptionChangeHandler> mHandlers = new ConcurrentHashMap<>();
 
     // transport Notification listener that will process subscription change notifications
     private final UListener mNotificationListener = this::handleNotifications;
-    
+
+
+    /**
+     * Creates a new USubscription client passing only a transport.
+     * 
+     * @param transport the transport to use for sending the notifications
+     */
+    public InMemoryUSubscriptionClient (UTransport transport) {
+        this(transport, CallOptions.DEFAULT);
+    }
+
+    /**
+     * Creates a new USubscription client passing {@link UTransport} and {@link CallOptions}
+     * used to provide additional options for the RPC requests to uSubscription service.
+     * 
+     * @param transport the transport to use for sending the notifications
+     * @param options the call options to use for the RPC requests
+     */
+    public InMemoryUSubscriptionClient (UTransport transport, CallOptions options) {
+        this(transport, new InMemoryRpcClient(transport), new SimpleNotifier(transport), options);
+    }
+
 
     /**
-     * Creates a new subscriber for existing Communication Layer client implementations.
+     * Creates a new USubscription client passing {@link UTransport}, {@link RpcClient} and {@link Notifier}.
      * 
      * @param transport the transport to use for sending the notifications
      * @param rpcClient the rpc client to use for sending the RPC requests
      * @param notifier the notifier to use for registering the notification listener
+     * @param options the call options to use for the RPC requests
      */
-    public InMemorySubscriber (UTransport transport, RpcClient rpcClient, Notifier notifier) {
+    public InMemoryUSubscriptionClient (UTransport transport, RpcClient rpcClient, Notifier notifier) {
+        this(transport, rpcClient, notifier, CallOptions.DEFAULT);
+    }
+
+
+    /**
+     * Creates a new USubscription client passing {@link UTransport}, {@link CallOptions},
+     * and an implementation of {@link RpcClient} and {@link Notifier}.
+     * 
+     * @param transport the transport to use for sending the notifications
+     * @param rpcClient the rpc client to use for sending the RPC requests
+     * @param notifier the notifier to use for registering the notification listener
+     * @param options the call options to use for the RPC requests
+     */
+    public InMemoryUSubscriptionClient (UTransport transport, RpcClient rpcClient,
+        Notifier notifier, CallOptions options) {
```

Comment:
CallOptions contains auth token and it is immutable. When token expires it has to be refreshed, so the current implementation will require to rebuild the whole instance just to update token, I think it is better to pass options into each RPC method.

Reply:
ok fixed